### PR TITLE
Update documentation with clearer instructions for setup and deployment

### PR DESCRIPTION
**Pull request by thefireblade**
# Description
Updating the README with relevant information on setup requirements including what kind of system is needed in order to deploy a Tesla Fleet Telemetry server as well updating documentation flow for setting up certificates, keys, and configuration files.

Also adding examples on how to run the server through Docker and manually.

## Type of change

Please select all options that apply to this change:

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation update

# Checklist:

Confirm you have completed the following steps:

- [x] My code follows the style of this project.
- [x] I have performed a self-review of my code.
- [x] I have made corresponding updates to the documentation.
- [x] I have added/updated unit tests to cover my changes.
- [x] I have added/updated integration tests to cover my changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,21 +9,42 @@ Fleet Telemetry is a server reference implementation for Tesla's telemetry proto
 
 The service handles device connectivity as well as receiving and storing transmitted data. Once configured, devices establish a WebSocket connection to push configurable telemetry records. Fleet Telemetry provides clients with ack, error, or rate limit responses.
 
-## Configuring and running the service
+## Prerequisites
+There are a few system prerequisites/recommendations for installing and deploying a Fleet Telemetry server. It is highly recommended that you run this service in a Debian server. In order for any Fleet Telemetry server to interact with vehicles, you will need to own a website domain and host Fleet Telemetry on that domain, which is defined as the `partner_domain` in the [Fleet API documentation](https://developer.tesla.com/docs/fleet-api#fleet-telemetry).
```
Comment (ThomasAlxDmy) [inline README.md:13]: For ease of installation the recommendation is to run this service in a Debian server

File: README.md [hunk 2]
```diff
@@ -9,21 +9,42 @@ Fleet Telemetry is a server reference implementation for Tesla's telemetry proto
...
+* Kubernetes/Helm*
 
-### Install manually (skip this if you have installed with Helm on Kubernetes)
-1. Allocate and assign a [FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name). This will be used in the server and client (vehicle) configuration.
+*Docker and Kubernetes/Helm are required for installing through deploying an image or using Helm Charts (recommended).
 
-2. Design a simple hosting architecture. We recommend: Firewall/Loadbalancer -> Fleet Telemetry -> Kafka.
+### Vehicle Compatibility
+
+Vehicles must be running firmware version 2023.20.6 or later. You may find this information under your VIN in the Tesla Mobile App or the Software tab in the vehicle's infotainment system. Some older model S/X are not supported.
```
Comment (ThomasAlxDmy) [inline README.md:34]: @aaronpkahn we might want to drop a line here about the new self-service version.

File: README.md (hunk 1 above)
Comment (thefireblade) [reply README.md]: Updated

**Review by agbpatro — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -9,21 +9,40 @@ Fleet Telemetry is a server reference implementation for Tesla's telemetry proto
...
-### Install on Kubernetes with Helm Chart (recommended)
-For ease of installation and operation, we recommend running Fleet Telemetry on Kubernetes. Helm Charts help you define, install, and upgrade applications on Kubernetes. You can find a reference helm chart [here](https://github.com/teslamotors/helm-charts/blob/main/charts/fleet-telemetry/README.md).
+* Go 1.20+
+* libzmq
+* Docker*
+* Kubernetes/Helm*
 
-### Install manually (skip this if you have installed with Helm on Kubernetes)
-1. Allocate and assign a [FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name). This will be used in the server and client (vehicle) configuration.
+*Docker and Kubernetes/Helm are required for installing through deploying an image or using Helm Charts (recommended).
```
[the remark this answers is not in the document]
Comment (thefireblade) [reply README.md:30]: sure

[the remark this answers is not in the document]
Comment (thefireblade) [reply README.md:30]: Not if I don't want it to convert to a bullet point

File: README.md [hunk 4]
```diff
@@ -9,21 +9,40 @@ Fleet Telemetry is a server reference implementation for Tesla's telemetry proto
...
+### Vehicle Compatibility
+
+Vehicles must be running firmware version 2023.20.6 or later. You may find this information under your VIN in the Tesla Mobile App or the Software tab in the vehicle's infotainment system. Some older model S/X are not supported.
+
+## Configuring and running the service
+As mentioned in prerequisites, you will need to own a public domain that will run this server in order for the vehicles to communicate with your Fleet Telemetry server. You may generate [self-signed certificates](https://en.wikipedia.org/wiki/Self-signed_certificate) for your domain and the generated tls certificate + private key pair will be used for the server to authenticate vehicles connecting as well as a means for devices to connect to your Fleet Telemetry server. Tesla devices will rely on a mutual TLS (mTLS) WebSocket to create a connection with the backend. Fleet Telemetry has been designed to operate on top of Kubernetes, but you can run it as a standalone binary if you prefer.
 
-3. Ensure mTLS connections are terminated on the Fleet Telemetry service.
+You may generate self a self-signed certificate using OpenSSL (have your domain/CNAME ready):
```
[the remark this answers is not in the document]
Comment (thefireblade) [reply README.md]: Updated but I will still include `(have your domain/CNAME ready)`

File: README.md [hunk 5]
```diff
@@ -9,21 +9,40 @@ Fleet Telemetry is a server reference implementation for Tesla's telemetry proto
 
 The service handles device connectivity as well as receiving and storing transmitted data. Once configured, devices establish a WebSocket connection to push configurable telemetry records. Fleet Telemetry provides clients with ack, error, or rate limit responses.
 
-## Configuring and running the service
+## Prerequisites
+There are a few system prerequisites/recommendations for installing and deploying a Fleet Telemetry server. For ease of installation, the recommendation is to run this service in a Debian server. In order for any Fleet Telemetry server to interact with vehicles, you will need to own a website domain and host Fleet Telemetry on that domain.
```
[the remark this answers is not in the document]
Comment (thefireblade) [reply README.md]: Updated

File: README.md [hunk 6]
```diff
@@ -9,21 +9,38 @@ Fleet Telemetry is a server reference implementation for Tesla's telemetry proto
...
-For ease of installation and operation, we recommend running Fleet Telemetry on Kubernetes. Helm Charts help you define, install, and upgrade applications on Kubernetes. You can find a reference helm chart [here](https://github.com/teslamotors/helm-charts/blob/main/charts/fleet-telemetry/README.md).
+Vehicles must be running firmware version 2023.20.6 or later. You may find this information under your VIN in the Tesla Mobile App or the Software tab in the vehicle's infotainment system. Some older model S/X are not supported.
 
-### Install manually (skip this if you have installed with Helm on Kubernetes)
-1. Allocate and assign a [FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name). This will be used in the server and client (vehicle) configuration.
+## Configuring and running the service
+You may generate [self-signed certificates](https://en.wikipedia.org/wiki/Self-signed_certificate) for your domain and the generated tls certificate + private key pair will be used to authenticate vehicles to your Fleet Telemetry server. Tesla vehicles rely on a mutual TLS (mTLS) WebSocket to create a connection with the backend.
```
Comment (aaronpkahn) [inline README.md:35]: please file a follow up to remove all instances of `you` `your` etc

Comment (thefireblade) [reply README.md:35]: Done

File: README.md [hunk 7]
```diff
@@ -9,21 +9,38 @@ Fleet Telemetry is a server reference implementation for Tesla's telemetry proto
 
 The service handles device connectivity as well as receiving and storing transmitted data. Once configured, devices establish a WebSocket connection to push configurable telemetry records. Fleet Telemetry provides clients with ack, error, or rate limit responses.
 
-## Configuring and running the service
+## Requirements
+These are the minimum system requirements for a Fleet Telemetry server, with 10 active vehicles connected, running on a Debian-based server with regular day-to-day usage. The requirements will change based on vehicle usage and quantity of vehicles.
```
Comment (aaronpkahn) [inline README.md:13]: minor: the first comma here isn't grammatically correct.
Maybe reword to something like: `These minimum system requirements assume a server with: 10 active vehicles connected, running on a Debian-based image, and with regular day-to-day vehicle usage.`

File: README.md [hunk 8]
```diff
@@ -9,21 +9,38 @@ Fleet Telemetry is a server reference implementation for Tesla's telemetry proto
...
-2. Design a simple hosting architecture. We recommend: Firewall/Loadbalancer -> Fleet Telemetry -> Kafka.
+You may generate a self-signed certificate using the OpenSSL command below, though we recommend using best practices for generating a server certificate from a trusted certificate authority (have your domain/CNAME ready):
 
-3. Ensure mTLS connections are terminated on the Fleet Telemetry service.
+```sh
+openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days 365 -out certificate.pem
+```
 
-4. Configure the server
+Before you deploy your Fleet Telemetry server, you will need to configure your server (the docker image will check the config mounted on `/etc/fleet-telemetry/config.json`). Server config template:
```
Comment (aaronpkahn) [inline README.md:43]: Suggested rewordding: `Create a server configuration, which by default is mounted on `/etc/fleet-telemetry/config.json. Template:`

File: README.md [hunk 9]
```diff
@@ -77,7 +94,28 @@ For ease of installation and operation, we recommend running Fleet Telemetry on
 ```
 Example: [server_config.json](./examples/server_config.json)
 
-5. Deploy and run the server. Get the latest docker image information from our [docker hub](https://hub.docker.com/r/tesla/fleet-telemetry/tags). This can be run as a binary via `./fleet-telemetry -config=/etc/fleet-telemetry/config.json` directly on a server, or as a Kubernetes deployment. Example snippet:
+### Deploy using Kubernetes with Helm Charts (recommended for large fleets)
+Hosting on [Kubernetes](https://kubernetes.io/) will enable the service to scale for large fleets. Helm Charts help you define, install, and upgrade applications on Kubernetes. Reference Helm chart [here](https://github.com/teslamotors/helm-charts/blob/main/charts/fleet-telemetry/README.md).
+
+You must have [Kubernetes](https://kubernetes.io/docs/setup/) amd [Helm](https://helm.sh/docs/intro/install/) installed for this deployment method.
```
Comment (aaronpkahn) [inline README.md:100]: reword: "Requires installation of [Kubernetes](https://kubernetes.io/docs/setup/) and [Helm](https://helm.sh/docs/intro/install/).

File: README.md (hunk 7 above)
Comment (ThomasAlxDmy) [reply README.md:13]: Why are we having 10 active vehicle as a reference? probably better to have 1k.

File: README.md [hunk 10]
```diff
@@ -77,7 +94,28 @@ For ease of installation and operation, we recommend running Fleet Telemetry on
...
+
+### Deploy using Docker
+
+1. Pull the Tesla Fleet Telemetry image (you may find all images in [docker hub](https://hub.docker.com/r/tesla/fleet-telemetry/tags)):
```
Comment (aaronpkahn) [inline README.md:104]: reword:  "Pull the Tesla Fleet Telemetry image. Images are located in [docker hub](https://hub.docker.com/r/tesla/fleet-telemetry/tags)):"

File: README.md [hunk 11]
```diff
@@ -77,7 +94,28 @@ For ease of installation and operation, we recommend running Fleet Telemetry on
...
+```sh
+docker pull tesla/fleet-telemetry:v0.1.8
+```
+2. Run and deploy your docker image with your mTLS certificate, private key, and config.json locally mounted on `/etc/fleet-telemetry`:
```
Comment (aaronpkahn) [inline README.md:108]: reword: "Run and deploy the docker image with an mTLS certificate, private key, and config.json locally mounted on `/etc/fleet-telemetry`"

File: README.md [hunk 12]
```diff
@@ -77,7 +94,28 @@ For ease of installation and operation, we recommend running Fleet Telemetry on
 ```
 Example: [server_config.json](./examples/server_config.json)
 
-5. Deploy and run the server. Get the latest docker image information from our [docker hub](https://hub.docker.com/r/tesla/fleet-telemetry/tags). This can be run as a binary via `./fleet-telemetry -config=/etc/fleet-telemetry/config.json` directly on a server, or as a Kubernetes deployment. Example snippet:
+### Deploy using Kubernetes with Helm Charts (recommended for large fleets)
+Hosting on [Kubernetes](https://kubernetes.io/) will enable the service to scale for large fleets. Helm Charts help you define, install, and upgrade applications on Kubernetes. Reference Helm chart [here](https://github.com/teslamotors/helm-charts/blob/main/charts/fleet-telemetry/README.md).
```
Comment (aaronpkahn) [inline README.md:98]: `Helm Charts help you define` -> `Helm Charts are used to define`

Comment (aaronpkahn) [reply README.md:98]: Also, inconsistent capitalization: `Reference Helm chart` -> `Reference Helm Chart`? Or make chart lowercase in all examples?

File: README.md (hunk 6 above)
Comment (Bre77) [inline README.md:35]: You can use a self signed certificate to start the server, but `/api/1/vehicles/fleet_telemetry_config` will reject anything not signed by Tesla's CA. I do not believe this line is accurate in my experience.

Comment (thefireblade) [reply README.md:35]: Can you post what error you are receiving? I used a self-signed certificate for the fleet-telemetry server and it should be working as long as it registers properly with the endpoint.

Comment (Bre77) [reply README.md:35]: > Can you post what error you are receiving? I used a self-signed certificate for the fleet-telemetry server and it should be working as long as it registers properly with the endpoint.

`tesla_fleet_api.exceptions.Forbidden: {'response': None, 'error': 'Please follow instructions https://developer.tesla.com/docs/fleet-api#fleet-telemetry to send CSR to Tesla', 'error_description': ''}`

Comment (agbpatro) [reply README.md:35]: Hi Brett
The error is because as discussed https://github.com/teslamotors/fleet-telemetry/issues/41#issuecomment-1875856991. 
We are onboarding new csr submissions incrementally.

Comment (Bre77) [reply README.md:35]: > Hi Brett
> The error is because as discussed https://github.com/teslamotors/fleet-telemetry/issues/41#issuecomment-1875856991. 
> We are onboarding new csr submissions incrementally. 

That's correct, so I am trying to point out that documentation that says you can use a self signed certificate is incorrect.

Comment (agbpatro) [reply README.md:35]: Hmm I don't think thats the issue. If you have mailed your CSR and haven't heard back, that means your account isn't ready for step 3 as mentioned in https://developer.tesla.com/docs/fleet-api#fleet-telemetry
